### PR TITLE
implemented monochrome ui, binary game mechanic change and formatted …

### PR DESCRIPTION
**Pull request by DexterStorey**
A very opinionated ui branch that could use some work / review before merging

Screen Shot 2022-10-22 at 11 59 12 AM
Screen Shot 2022-10-22 at 11 57 57 AM

### CODE REVIEW

**Comment by DexterStorey** [comment]
I feel like we need to decide on boldness, roundness and font sizing as part of a design system... I made some arbitrary choices here

**Comment by DexterStorey** [comment]
should the keys be square? also just realizing the padding's inconsistent between the grid and the keyboard

**Comment by DexterStorey** [comment]
Also wondering if the indicators on top / bottom would look cleaner as small arrows that hug a bit closer to the char

**Comment by tedspare** [comment]
ITYJ re design system 😅 

---

> Also wondering if the indicators on top / bottom would look cleaner as small arrows that hug a bit closer to the char

Core issue here: does an up/high indicator mean "guess higher" or "you're too high"?
- One approach is to replace "high vs low" with "close vs far". This would make binary search 2x harder too.
- Another is to add a legend: `^ = guess higher`

File: src/pages/index.tsx [hunk 1]
```diff
@@ -15,20 +25,30 @@ const hexes = Array(16)
...
-      ? "bg-blue-600/50"
-      : distance < 0
-      ? "bg-blue-600/30"
-      : distance == 0
-      ? "bg-green-600/50"
-      : distance > 5
-      ? "bg-red-600/50"
-      : "bg-red-600/30";
-  return hotColdColor;
+  return distance < 0 ? HINT.UP : distance == 0 ? HINT.HIT : HINT.DOWN;
```
Comment (tedspare) [inline src/pages/index.tsx:28]: I like this

File: src/pages/index.tsx [hunk 2]
```diff
@@ -188,32 +233,36 @@ const Home = ({ hex }: IProps) => {
           </div>
           {/* Keyboard */}
           <div className="center max-w-xl flex-wrap gap-2">
-            {hexes.map((hex: string, i: number) => (
+            {hexes.map((_hex: string, i: number) => (
```
Comment (tedspare) [inline src/pages/index.tsx:236]: `_hex` -> `char` maybe? For clarity

File: src/pages/index.tsx [hunk 3]
```diff
@@ -188,32 +233,36 @@ const Home = ({ hex }: IProps) => {
...
+                style={{ color: `#${hex}`, backgroundColor: contrastColor }}
               >
-                <span>{hex}</span>
+                <span>{_hex}</span>
               </div>
             ))}
             <div
               onClick={() => handleBackspace(currentRow, currentColumn)}
               className="keyboard px-2"
+              style={{ color: `#${hex}`, backgroundColor: contrastColor }}
```
Comment (tedspare) [inline src/pages/index.tsx:249]: Could parametrize `const color = '#${hex}'` above

File: src/styles/globals.css [hunk 4]
```diff
@@ -3,22 +3,20 @@
...
+  p,
+  a {
+    @apply font-medium;
   }
   .center {
     @apply flex items-center justify-center;
   }
   .keyboard {
-    @apply center h-16 w-12 rounded bg-gray-700 transition-transform hover:scale-110 hover:bg-gray-600;
+    @apply center h-16 w-12 cursor-pointer font-bold transition-transform hover:scale-110;
```
Comment (tedspare) [inline src/styles/globals.css:20]: `transition-transform` -> `transition-all` for the hover color

**Review by tedspare — CHANGES_REQUESTED** [review]
Generally LGTM; nitpicks only

**Comment by DexterStorey** [comment]
> LGTM
R u even a FAANG PM

**Comment by DexterStorey** [comment]
> Generally LGTM; nitpicks only

Check it out I think this is ready to go

**Comment by DexterStorey** [comment]
Looks good on small mobile

image

**Comment by DexterStorey** [comment]
@tedspare Can you review?

Need help testing the social media preview and Fav coverage on your browsers

Otherwise I think this is good to go- I'll make new prs for styling fine tuning later but I want to merge so I can branch off main for some of the new features

**Comment by DexterStorey** [comment]
TLDR there's some big ui changes with spacing and sizing - it's very mobile first, but desktop could use a little love when we get a chance

File: src/pages/api/favicon.tsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          style={{
+            backgroundColor: bg,
+            height: "100%",
+            width: "100%",
+            display: "flex",
+            textAlign: "center",
+            alignItems: "center",
+            justifyContent: "center",
+            flexDirection: "column",
+            flexWrap: "nowrap",
```
Comment (tedspare) [inline src/pages/api/favicon.tsx:29]: you really hate Tailwind eh

Comment (tedspare) [reply src/pages/api/favicon.tsx:29]: nvm this is the favicon hoooly

File: src/pages/api/og.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
+import { ImageResponse } from "@vercel/og";
+import { NextRequest } from "next/server";
+
+export const config = {
+  runtime: "experimental-edge",
```
Comment (tedspare) [inline src/pages/api/og.tsx:5]: ❤️

**Comment by tedspare** [comment]
@DexterStorey LGTM 🚢 🚢 🚢

**Comment by tedspare** [comment]
Favicon in Arc browser looks good
image

Social preview in Slack is a different color but looks good
image

Lost (I suck at this game so far lol) but didn't get the alert
image

Also keyboard doesn't show `active:` animations anymore - intentional?

**Comment by DexterStorey** [comment]
> Also keyboard doesn't show `active:` animations anymore - intentional?

removed with the monochrome ui - unless you can think of a way to show active that doesn't involve transparency / color? - it seems too quirky to make it bounce or invert..

Maybe we can make this a new issue about broader monochrome constraints and then think about ways to handle multiple states and notifications with monochrome ui?

**Comment by DexterStorey** [comment]
We also need to design monochrome ui for alerts / win / loss conditions but I'll make this an issue and self assign

**Comment by DexterStorey** [comment]
F it I'm merging

This merge closes issues #4 and #5 

#8 and #9 are created as a result of the breaking changes in this branch
